internal/provider/pipeline_resource.go

```diff
@@ -0,0 +1,289 @@
+// pipelineModel maps credentials schema data.
+type pipelineModel struct {
+	VirtualClusterID  types.String    `tfsdk:"virtual_cluster_id"`
+	ID                types.String    `tfsdk:"id"`
+	Name              types.String    `tfsdk:"name"`
+	State             types.String    `tfsdk:"state"`
+	ConfigurationYAML utils.YamlValue `tfsdk:"configuration_yaml"`
+	ConfigurationID   types.String    `tfsdk:"configuration_id"`
+}
+
+// Configure adds the provider configured client to the data source.
+func (r *pipelineResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*api.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Invalid Provider Configuration",
+			fmt.Sprintf("Expected an API Client instance, but got: %T. Please ensure the provider is configured correctly.", req.ProviderData),
+		)
+		return
+	}
+
+	r.client = client
+}
+
+// Metadata returns the resource type name.
+func (r *pipelineResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_pipeline"
+}
+
+// Schema defines the schema for the resource.
+func (r *pipelineResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: `
+This resource allows you to create pipelines.
+For more details, take a look at: https://docs.warpstream.com/warpstream/configuration/benthos
+`,
+		Attributes: map[string]schema.Attribute{
+			"virtual_cluster_id": schema.StringAttribute{
+				Description: "The ID of the virtual cluster associated with the pipeline.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "The unique human-readable name of the pipeline within the virtual cluster. This cannot be changed after creation.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"id": schema.StringAttribute{
+				Description: "The unique identifier of the pipeline, automatically generated by WarpStream upon creation.",
+				Computed:    true,
+			},
+			"state": schema.StringAttribute{
+				Description: "The desired operational state of the pipeline. Valid values are 'running' or 'paused'.",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("running", "paused"),
+				},
+			},
+			"configuration_yaml": schema.StringAttribute{
+				Description: "The YAML content defining the input sources, processing steps, and output destinations for the pipeline. " +
+					"This represents the complete configuration for this specific version. To understand how to set your configuration take a look at: https://docs.warpstream.com/warpstream/configuration/benthos#getting-started",
```

Comment:
I opened a gitbook diff to rename this slug configuration/bento. We might want to merge that and update this line before merging this pr.

https://app.gitbook.com/o/xuJtdDASiqpJSjc2Fs19/s/jB7FxO8ty4EXO4HsQP4E/~/changes/408

Reply:
Actually I closed that diff and opened this pr which is more comprehensive https://github.com/warpstreamlabs/docs/pull/72